src/static/app.js

```diff
@@ -84,3 +108,42 @@ document.addEventListener("DOMContentLoaded", () => {
   // Initialize app
   fetchActivities();
 });
+
+// Function to cancel registration
+async function cancelRegistration(activityName, email) {
+  const messageDiv = document.getElementById("message");
+  
+  try {
+    const response = await fetch(
+      `/activities/${encodeURIComponent(activityName)}/cancel?email=${encodeURIComponent(email)}`,
+      {
+        method: "DELETE",
+      }
+    );
+
+    const result = await response.json();
+
+    if (response.ok) {
+      messageDiv.textContent = result.message;
+      messageDiv.className = "success";
+      
+      // Refresh the activities display
+      await fetchActivities();
+    } else {
+      messageDiv.textContent = result.detail || "An error occurred";
+      messageDiv.className = "error";
+    }
+
+    messageDiv.classList.remove("hidden");
+
+    // Hide message after 5 seconds
+    setTimeout(() => {
+      messageDiv.classList.add("hidden");
+    }, 5000);
+  } catch (error) {
+    messageDiv.textContent = "Failed to cancel registration. Please try again.";
+    messageDiv.className = "error";
+    messageDiv.classList.remove("hidden");
+    console.error("Error cancelling registration:", error);
+  }
+}
```

Comment:
成功或失败的消息处理合并到一起，使代码更加简洁易读
```suggestion
async function cancelRegistration(activityName, email) {
  const messageDiv = document.getElementById("message");
  try {
    const response = await fetch(
      `/activities/${encodeURIComponent(activityName)}/cancel?email=${encodeURIComponent(email)}`,
      { method: "DELETE" }
    );
    const result = await response.json();
    messageDiv.textContent = response.ok ? result.message : (result.detail || "An error occurred");
    messageDiv.className = response.ok ? "success" : "error";
    messageDiv.classList.remove("hidden");
    if (response.ok) await fetchActivities();
  } catch (error) {
    messageDiv.textContent = "Failed to cancel registration. Please try again.";
    messageDiv.className = "error";
    messageDiv.classList.remove("hidden");
    console.error("Error cancelling registration:", error);
  }
  setTimeout(() => messageDiv.classList.add("hidden"), 5000);
}
```